clash-prelude/src/Clash/Sized/Vector/Par.hs

```diff
@@ -58,7 +58,8 @@ scanlInductiveRTree
   -> RTree n a
   -> Vec (2^n) a
 scanlInductiveRTree op tr =
-  -- I have to use sameNat and compareSNat both; otherwise it doesn't compile
+  -- I have to use sameNat and compareSNat both; the proofs that n>=0 and n~0
```

Comment:
To be pedantic, with the way you're using `compareSNat` you end up with a proof for `1 <= n`, instead of a proof for `n >= 0` since `SNatGT` has the type

```haskell
SNatGT :: forall a b. (b + 1) <= a => SNatLE a b
```

If you did want a proof `0 <= n` you would have to swap the arguments to `compareSNat`

Reply:
Corrected!